plugins/filter_record_modifier/filter_modifier.c

```diff
@@ -309,7 +370,14 @@ static int cb_modifier_filter(void *data, size_t bytes,
         kv = obj->via.map.ptr;
         for(i=0; bool_map[i] != TAIL_OF_ARRAY; i++) {
             if (bool_map[i] == TO_BE_REMAINED) {
-                msgpack_pack_object(&tmp_pck, (kv+i)->key);
+                char is_suffixed = FLB_FALSE;
+                if ( ctx->suffix_type && suffix_key_type(&tmp_pck, &(kv+i)->key, &(kv+i)->val) == FLB_TRUE) {
```

Comment:
please do not put a function call inside the if() conditional, instead do it before and use a int ret; to check the return value instead.